frame/evm/src/runner/contracts.rs

```diff
@@ -0,0 +1,278 @@
+		max_priority_fee_per_gas: Option<U256>,
+		nonce: Option<U256>,
+		access_list: Vec<(H160, Vec<H256>)>,
+		is_transactional: bool,
+		validate: bool,
+		weight_limit: Option<Weight>,
+		proof_size_base_cost: Option<u64>,
+		config: &evm::Config,
+	) -> Result<CallInfo, RunnerError<Self::Error>> {
+		if validate {
+			Self::validate(
+				source,
+				Some(target),
+				input.clone(),
+				value,
+				gas_limit,
+				max_fee_per_gas,
+				max_priority_fee_per_gas,
+				nonce,
+				access_list.clone(),
+				is_transactional,
+				weight_limit,
+				proof_size_base_cost,
+				config,
+			)?;
+		}
+		let origin = T::AddressMapping::into_account_id(source);
+		let dest = T::AddressMapping::into_account_id(target);
+		let (_base_fee, weight) = T::FeeCalculator::min_gas_price();
+		let value = value.try_into().map_err(|_| RunnerError {
+			error: Error::<T>::BalanceLow,
+			weight,
+		})?;
+		let ret = pallet_contracts::Pallet::<T>::bare_call(
+			origin,
+			dest,
+			value,
+			Weight::from_parts(gas_limit, u64::from(T::MaxCodeLen::get()) * 2),
+			None,
+			input,
+			pallet_contracts::DebugInfo::Skip,
+			pallet_contracts::CollectEvents::Skip,
+			pallet_contracts::Determinism::Enforced,
+		);
+		let retd = ret.result.map_err(|_| RunnerError {
+			error: Error::<T>::Undefined, // TODO: pallet contracts specific error.
+			weight: ret.gas_consumed,
+		})?;
+		let info = CallInfo {
+			exit_reason: ExitReason::Succeed(ExitSucceed::Stopped),
+			value: retd.data,
+			used_gas: UsedGas {
+				standard: ret.gas_consumed.ref_time().into(),
+				effective: ret.gas_consumed.ref_time().into(),
+			},
+			logs: Vec::new(), // TODO: we need to collect logs.
+			weight_info: Some(WeightInfo {
+				ref_time_limit: Some(ret.gas_required.ref_time()),
+				proof_size_limit: Some(ret.gas_required.proof_size()),
+				ref_time_usage: Some(ret.gas_consumed.ref_time()),
+				proof_size_usage: Some(ret.gas_consumed.proof_size()),
+			}),
+		};
+		Ok(info)
+	}
+
+	fn create(
+		source: H160,
+		init: Vec<u8>,
+		value: U256,
+		gas_limit: u64,
+		max_fee_per_gas: Option<U256>,
+		max_priority_fee_per_gas: Option<U256>,
+		nonce: Option<U256>,
+		access_list: Vec<(H160, Vec<H256>)>,
+		is_transactional: bool,
+		validate: bool,
+		weight_limit: Option<Weight>,
+		proof_size_base_cost: Option<u64>,
+		config: &evm::Config,
+	) -> Result<CreateInfo, RunnerError<Self::Error>> {
+		if validate {
+			Self::validate(
+				source,
+				None,
+				init.clone(),
+				value,
+				gas_limit,
+				max_fee_per_gas,
+				max_priority_fee_per_gas,
+				nonce,
+				access_list.clone(),
+				is_transactional,
+				weight_limit,
+				proof_size_base_cost,
+				config,
+			)?;
+		}
+		let (_base_fee, weight) = T::FeeCalculator::min_gas_price();
+		let (code, init_data, salt): (Vec<u8>, Vec<u8>, Vec<u8>) =
+			scale_codec::Decode::decode(&mut &init[..]).map_err(|_| RunnerError {
+				error: Error::<T>::Undefined, // TODO: pallet contracts specific error.
+				weight,
+			})?;
```

Comment:
This will work but only when we provide some custom tooling, right? Otherwise when using web3.js `init` will contain `init_args ++ code`, right?